Find a functionally equivalent code:com.android.systemui.recent.RecentTasksLoader.createTaskDescription:COMMENT
Method Modifier: 
Comment:// Create an TaskDescription, returning null if the title or icon is null

@@ -4,18 +4,23 @@
         intent.setComponent(origActivity);
     }
     final PackageManager pm = mContext.getPackageManager();
+    final IPackageManager ipm = AppGlobals.getPackageManager();
     intent.setFlags((intent.getFlags() & ~Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED) | Intent.FLAG_ACTIVITY_NEW_TASK);
-    final ResolveInfo resolveInfo = pm.resolveActivity(intent, 0);
+    ResolveInfo resolveInfo = null;
+    try {
+        resolveInfo = ipm.resolveIntent(intent, null, 0, userId);
+    } catch (RemoteException re) {
+    }
     if (resolveInfo != null) {
         final ActivityInfo info = resolveInfo.activityInfo;
         final String title = info.loadLabel(pm).toString();
         if (title != null && title.length() > 0) {
             if (DEBUG)
                 Log.v(TAG, "creating activity desc for id=" + persistentTaskId + ", label=" + title);
-            TaskDescription item = new TaskDescription(taskId, persistentTaskId, resolveInfo, baseIntent, info.packageName, description);
+            TaskDescription item = new TaskDescription(taskId, persistentTaskId, resolveInfo, baseIntent, info.packageName, description, userId);
             item.setLabel(title);
             return item;
         } else {

